### ISSUE TITLE
0.5.0: sphinx fails because missing path to `cssselect2` module

### ISSUE THREAD

**Issue by kloczek** [Kozea/cssselect2#28]
```console
+ /usr/bin/sphinx-build -n -T -b man docs build/sphinx/man
Running Sphinx v4.5.0

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/sphinx/config.py", line 332, in eval_config_file
    exec(code, namespace)
  File "/home/tkloczko/rpmbuild/BUILD/cssselect2-0.5.0/docs/conf.py", line 3, in <module>
    import cssselect2
ModuleNotFoundError: No module named 'cssselect2'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/sphinx/cmd/build.py", line 272, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
  File "/usr/lib/python3.8/site-packages/sphinx/application.py", line 202, in __init__
    self.config = Config.read(self.confdir, confoverrides or {}, self.tags)
  File "/usr/lib/python3.8/site-packages/sphinx/config.py", line 165, in read
    namespace = eval_config_file(filename, tags)
  File "/usr/lib/python3.8/site-packages/sphinx/config.py", line 345, in eval_config_file
    raise ConfigError(msg % traceback.format_exc()) from exc
sphinx.errors.ConfigError: There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/sphinx/config.py", line 332, in eval_config_file
    exec(code, namespace)
  File "/home/tkloczko/rpmbuild/BUILD/cssselect2-0.5.0/docs/conf.py", line 3, in <module>
    import cssselect2
ModuleNotFoundError: No module named 'cssselect2'


Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/sphinx/config.py", line 332, in eval_config_file
    exec(code, namespace)
  File "/home/tkloczko/rpmbuild/BUILD/cssselect2-0.5.0/docs/conf.py", line 3, in <module>
    import cssselect2
ModuleNotFoundError: No module named 'cssselect2'
```
This can be fixed by patch like below:
```patch
--- a/docs/conf.py~     2022-04-15 14:28:41.000000000 +0000
+++ b/docs/conf.py      2022-04-15 14:29:28.086804119 +0000
@@ -1,3 +1,7 @@
+import sys
+import os
+sys.path.append(os.path.abspath('..'))
+
 # cssselect2 documentation build configuration file.

 import cssselect2
```
With that patch sphinx still shows two warnings
```console
+ /usr/bin/python3 -sBm build -w --no-isolation
* Getting dependencies for wheel...
* Building wheel...
Successfully built cssselect2-0.5.0-py3-none-any.whl
+ /usr/bin/sphinx-build -n -T -b man docs build/sphinx/man
Running Sphinx v4.5.0
making output directory... done
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from https://pythonhosted.org/webencodings/objects.inv...
loading intersphinx inventory from https://doc.courtbouillon.org/tinycss2/latest/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [man]: all manpages
updating environment: [new config] 6 added, 0 changed, 0 removed
reading sources... [100%] support
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
writing... python-cssselect2.3 { first_steps api_reference changelog contribute support } /home/tkloczko/rpmbuild/BUILD/cssselect2-0.5.0/docs/first_steps.rst:33: WARNING: py:meth reference target not found: Matcher
/home/tkloczko/rpmbuild/BUILD/cssselect2-0.5.0/docs/first_steps.rst:35: WARNING: py:meth reference target not found: ElementWrapper
done
build succeeded, 2 warnings.
```

**Comment by liZe** [comment]
Hi.

`/usr/bin/sphinx-build -n -T -b man docs build/sphinx/man`

That’s strange, because Python can generally load modules in the current folder. If you generate documents with this command, you’re in the root folder of the repository, and the cssselect2 module is in this folder.

> With that patch sphinx still shows two warnings

Fixed in 003f3f8 (just a few minutes before your issue!).

**Comment by kloczek** [comment]
> That’s strange, because Python can generally load modules in the current folder

Nope .. it loads module from curent directory only when it is executed by `pyton -m foo_module` (look on python(1). man page where it is explained that it adds to `sys.path` current directory).
This is why calling pytest by `python -m pytest` is not proper way of using `pytest` and why pytest provides /usr/bin/pytest script. The same is with many other modules like `pip` ..

> Fixed in https://github.com/Kozea/cssselect2/commit/003f3f8cc01a25953971aaec36e03ca6be7d5152 (just a few minutes before your issue!).

Tested
```console
[tkloczko@devel-g2v SPECS]$ rpmbuild -ba python-cssselect2.spec
warning: Downloading https://github.com/Kozea/cssselect2//archive/0.5.0/python-cssselect2-0.5.0.tar.gz to /home/tkloczko/rpmbuild/SOURCES/python-cssselect2-0.5.0.tar.gz
warning: Downloading https://github.com/Kozea/cssselect2//commit/003f3f8c.patch#/python-cssselect2-Fix-documentation.patch to /home/tkloczko/rpmbuild/SOURCES/python-cssselect2-Fix-documentation.patch
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.y5DkTJ
+ umask 022
+ cd /home/tkloczko/rpmbuild/BUILD
+ cd /home/tkloczko/rpmbuild/BUILD
+ rm -rf cssselect2-0.5.0
+ /usr/bin/gzip -dc /home/tkloczko/rpmbuild/SOURCES/python-cssselect2-0.5.0.tar.gz
+ /usr/bin/tar -xof -
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ cd cssselect2-0.5.0
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ /usr/bin/patch -p1 -s --fuzz=0 --no-backup-if-mismatch -f
+ /usr/bin/cat /home/tkloczko/rpmbuild/SOURCES/python-cssselect2-man3_level.patch
+ /usr/bin/cat /home/tkloczko/rpmbuild/SOURCES/python-cssselect2-Fix-documentation.patch
+ /usr/bin/patch -p1 -s --fuzz=0 --no-backup-if-mismatch -f

[..]

+ /usr/bin/sphinx-build -n -T -b man docs build/sphinx/man
Running Sphinx v4.5.0
making output directory... done
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from https://pythonhosted.org/webencodings/objects.inv...
loading intersphinx inventory from https://doc.courtbouillon.org/tinycss2/latest/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [man]: all manpages
updating environment: [new config] 6 added, 0 changed, 0 removed
reading sources... [100%] support
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
writing... python-cssselect2.3 { first_steps api_reference changelog contribute support } done
build succeeded.
```
Now all is clean 😄 
Thank you 😃 
Closing.
